### PR TITLE
Add a new inspector for the component.

### PR DESCRIPTION
**Pull request by Nyan11**
Need Roassal to work.

image

You can clic on a component or a service and have it open in the second panel

image

To launch start an example and click on world menu > Molecule > inspect running component

### CODE REVIEW

**Comment by Nyan11** [comment]
@LabordePierre

**Comment by Nyan11** [comment]
There is currently a bug, the link does not really make sens right now.
I will fix it latter.

**Comment by labordep** [comment]
Thanks @Nyan11! Is it working with the default Roassal embedded into a Pharo image?

**Comment by Nyan11** [comment]
I think you need the full version of roassal.

We should make a variant in the baseline.

**Comment by labordep** [comment]
Can you add tests to check essentials?

**Comment by Nyan11** [comment]
Now layout look like this:
image


After replacing the component:
image

**Comment by Nyan11** [comment]
I have to move the class in a new package and create the baseline variant.

**Comment by labordep** [comment]
Are you sure that you are need a different roassal than the version in Pharo?

**Comment by Nyan11** [comment]
I just check, we do not need the full roassal version to be laoded in order to work.

**Comment by Nyan11** [comment]
It can be merged

**Comment by labordep** [comment]
Nice! I will review :-)

File: src/Molecule/MolHomeServices.class.st [hunk 1]
```diff
@@ -148,6 +148,19 @@ MolHomeServices >> initialize [
 	waitingForActivation := Set new
 ]
 
+{ #category : #'as yet unclassified' }
+MolHomeServices >> inspectionDeployedComponentsGraph [
```
Comment (labordep) [inline src/Molecule/MolHomeServices.class.st:152]: Is this method in IDE package?

File: src/Molecule/MolComponentImpl.trait.st [hunk 2]
```diff
@@ -202,6 +202,12 @@ MolComponentImpl classSide >> undeploy [
 	^ MolComponentManager default deploymentServices undeployComponentImplementation: self
 ]
 
+{ #category : #converting }
+MolComponentImpl >> asRSMoleculeShape [
```
Comment (labordep) [inline src/Molecule/MolComponentImpl.trait.st:206]: Is this method in IDE package?

File: src/Molecule/MolComponentImpl.trait.st [hunk 3]
```diff
@@ -418,6 +424,16 @@ MolComponentImpl >> forServices: aServicesTrait useProvider: aComponentName [
 		ifFalse: [ MolUtils log: self printString, '>forServices:useProvider: try to link a non services trait for services ' ]
 ]
 
+{ #category : #'as yet unclassified' }
+MolComponentImpl >> inspectionComponent [
```
Comment (labordep) [inline src/Molecule/MolComponentImpl.trait.st:428]: Is this method in IDE package?

File: src/Molecule/MolComponentImpl.trait.st (hunk 2 above)
Comment (Nyan11) [reply src/Molecule/MolComponentImpl.trait.st]: move to IDE

File: src/Molecule/MolHomeServices.class.st (hunk 1 above)
Comment (Nyan11) [reply src/Molecule/MolHomeServices.class.st]: move to IDE

File: src/Molecule/MolComponentImpl.trait.st (hunk 3 above)
Comment (Nyan11) [reply src/Molecule/MolComponentImpl.trait.st]: move to IDE

**Comment by Nyan11** [comment]
@LabordePierre @labordep Ready to merge

**Review by LabordePierre — APPROVED** [review]

**Pull request merged**